### PR TITLE
Added SearchPattern program

### PR DESCRIPTION
**Pull request by DavidJsson**

### CODE REVIEW

File: SearchPattern.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    public static void main(String[] args) throws FileNotFoundException {
+        if (args[0].equals("search")) {
+            SearchPattern search = new SearchPattern();
+            search.searchPattern(args[1], args[2]);
+        } else {
+            System.out.println("Invalid command");
+        }
+    }
+
+    public void searchPattern(String pattern, String file) throws FileNotFoundException {
```
Comment (JoePaul) [inline SearchPattern.java:16]: Might have been nice with some error handling

File: SearchPattern.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.util.Scanner;
+
+public class SearchPattern {
+
+    public static void main(String[] args) throws FileNotFoundException {
+        if (args[0].equals("search")) {
```
Comment (JoePaul) [inline SearchPattern.java:8]: Maybe ignore case for the command so it runs with both capital and lower case "s"

File: SearchPattern.java (hunk 1 above)
Comment (JoePaul) [inline SearchPattern.java:16]: Parameter "file" might be misleading in the method since its not the file that is being sent in but the path

Comment (JoePaul) [inline SearchPattern.java:16]: The function name could be a bit clearer on what it actually does

**Review by JoePaul — COMMENTED** [review]
Nice code with some minor things that could be changed and some that's just a heads up. No major issues :)

File: SearchPattern.java (hunk 1 above)
Comment (DavidJsson) [reply SearchPattern.java]: Added some error handling.

File: SearchPattern.java (hunk 2 above)
Comment (DavidJsson) [reply SearchPattern.java:8]: The command should be defined as lowercase search.

File: SearchPattern.java (hunk 1 above)
Comment (DavidJsson) [reply SearchPattern.java]: File changed to path to make clearer.

Comment (DavidJsson) [reply SearchPattern.java]: Function name is elaborated to give a clearer context.

File: SearchPattern.java [hunk 3]
```diff
@@ -4,22 +4,27 @@
...
 
-    public static void main(String[] args) throws FileNotFoundException {
+    public static void main(String[] args) {
         if (args[0].equals("search")) {
             SearchPattern search = new SearchPattern();
-            search.searchPattern(args[1], args[2]);
+            search.searchPatternsInFile(args[1], args[2]);
         } else {
             System.out.println("Invalid command");
         }
```
Comment (JoePaul) [inline SearchPattern.java]: No error handling when passing in too few or too many arguments to the function

Comment (JoePaul) [reply SearchPattern.java]: Missed this the first review! Sorry!

Comment (DavidJsson) [reply SearchPattern.java]: Error handling added.

**Review by JoePaul — APPROVED** [review]
Nice job!

**Pull request merged**